File: Data Structures & Algorithms/max-area-of-island/submission-3.py
class Solution:
    def maxAreaOfIsland(self, grid: List[List[int]]) -> int:
        rows = len(grid)
        cols = len(grid[0])
        maxArea = 0
        directions = [(-1, 0), (1, 0), (0, -1), (0, 1)]

        def dfs(row, col):
            if (row < 0 or row == rows or
                col < 0 or col == cols or 
                not grid[row][col] or grid[row][col] == 0):
                return 0
            
            grid[row][col] = 0

            area = 1 + (dfs(row + 1, col) +
                dfs(row - 1, col) +
                dfs(row, col + 1) +
                dfs(row, col - 1))
            return area

        for row in range(rows):
            for col in range(cols):
                if grid[row][col] == 1:
                    maxArea = max(maxArea, dfs(row, col))
        
        return maxArea

File: Data Structures & Algorithms/climbing-stairs/submission-3.js
class Solution {
    /**
     * @param {number} n
     * @return {number}
     */
    climbStairs(n) {
        const dfs = (i) => {
            if (i >= n) {
                return i === n ? 1 : 0;
            }

            const a = dfs(i + 1);
            const b = dfs(i + 2);

            // console.log(a, b)

            return a + b;
        }

        console.log("dfs(0)",dfs(0))

        return dfs(0);
    }
}
